cy43439.go

```diff
@@ -514,6 +595,12 @@ f2ready:
 		return err
 	}
 
+	//
+	// Start async polling to service ioctls, events, and data
+	//
+
+	d.pollStart()
```

Comment:
pollStart here? Doesen't polling interfere with SPI, which I believe is done by the functions below like `clmLoad`

Reply:
SPI transactions should be done under d.mu.Lock().  Polling is done with Lock held, and sendIotcl() holds the Lock, so both aren't running at the same time.  Do you see any other paths where SPI transactions would happen unlocked?

Reply:
The final place where we'll need to hold the Lock is in the Tx path.  To summarize, the Lock is held in three cases:

SendEth()
sendIotcl()
poll()

Reply:
poll() sleeps 5ms with Lock not held so SendEth() and sendIoctl() can make progress.  Not sure what a good sleep value would be.   Its value affects Rx latency.  poll() is handling Rx, so any delay sleeping in poll() is adding to latency on Rx.  I guess you could say, on average, the driver introduces a (5/2)ms latency on Rx packets.

Reply:
Since we'd eventually setup a interrupt to listen for a network event the polling would inadvertently trigger during a SPI transaction even if the lock was held. We should design the API so that we can swap in/out different implementations I feel.

Reply:
I think we're OK.

When we switch from polling to irq, the irq handler wouldn't try to grab the lock.  Minimally, the irq handler would do just two things, not under lock:

```
func (d *Device) irqhandler() {
    // ACK irq
    go d.softIrq()
}
```

Where softIrq() will grab the lock:

```
func (d *Device) softIrq() {
    d.hw.Lock()
    defer d.hw.Unlock()
    d.processPackets()
}
```

Basically switch from HARD IRQ context to SOFT IRQ context using a goroutine.